### PR TITLE
refactor: Updates to exclusively create a VPC network

### PR DESCRIPTION
**Pull request by moonshot-s**
Updates to networking module (variables and main) to be able to 
* exclusively create a VPC network.
* private_ip_google_access = true, When enabled, VMs in this subnetwork without external IP addresses can access Google APIs and services by using Private Google Access.

### CODE REVIEW

File: modules/networking/main.tf [hunk 1]
```diff
@@ -3,6 +3,10 @@ resource "google_compute_network" "default" {
   name                    = "${var.namespace}-vpc"
   description             = "${var.namespace} VPC Network"
   auto_create_subnetworks = false
+  project = var.project
```
Comment (BBischof) [inline modules/networking/main.tf]: it seems like different linting is being used? is there pre-commit for this repo?

Comment (chrishiste) [reply modules/networking/main.tf]: There is, @moonshot-s please make sure you have configured your local development environment: https://github.com/wandb/terraform-google-dagster#development

**Review by chrishiste — COMMENTED** [review]
Please lint the code. And make sure to get @zythosec approval too.

**Review by zythosec — APPROVED** [review]

**Comment by jsbroks** [comment]
This PR is included in version 1.1.0 :tada:

**Pull request merged**